File: tools/gyp/test/win/linker-flags/inline_test.cc
// Copyright (c) 2013 Google Inc. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "inline_test.h"

#include <intrin.h>
#pragma intrinsic(_ReturnAddress)

bool IsFunctionInlined(void* caller_return_address) {
  return _ReturnAddress() == caller_return_address;
}

File: deps/v8/test/js-perf-test/RegExp/inline_test.js
// Copyright 2018 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

function ConstTest() {
  /[Cz]/.test("abCdefgzabCdefgzabCdefgzabCdefgzabCdefgzabCdefgzabCdefgzabCdefgz");
}

const cre = /[Cz]/;
function GlobalConstTest() {
  cre.test("abCdefgzabCdefgzabCdefgzabCdefgzabCdefgzabCdefgzabCdefgzabCdefgz");
}

var benchmarks = [ [ConstTest, () => {}],
                   [GlobalConstTest, () => {}],
                 ];
createBenchmarkSuite("InlineTest");
